### ISSUE TITLE
[version request]: Firefox port

### ISSUE THREAD

**Issue by Majoramari** [richardadonnell/Upwork-Job-Scraper#86]

**Comment by richardadonnell** [comment]
thank you for requesting this, and apologies for the delayed response. I've been thinking seriously about completely refactoring this extension, so it can be cross-browser compatible (Chromium + Firefox) - but, the last 3 months of the year are always crazy busy for me. I'll have more time to think about this in a few weeks ...

note: as of November 2024, Firefox only has a 6.45% share of the worldwide browser market. more info: https://gs.statcounter.com/browser-market-share/desktop/worldwide

As much as I love firefox, (been using it since it launched) the userbase is _literally_ 1/10th the size compared to chrome or chromium-based browsers - and more broadly speaking, the recent (and also current) state and future direction of the Mozilla corporation.

tl;dr: I'll consider it, but it will be at least a few weeks, until I can even begin to consider it. 🍻

**Comment by richardadonnell** [comment]
Thanks so much for your patience on this.
I wanted to share a quick update on the Firefox port request.
The extension has been completely refactored from scratch and now runs on WXT.
The Firefox port build process has been tested and is working.
Before release, we still need to finish functional validation in Firefox.
If all checks look good, you should expect a Firefox release soon.
Appreciate you sticking with us!

**Comment by richardadonnell** [comment]
<img width="1214" height="118" alt="Image" src="https://github.com/user-attachments/assets/e110199d-340b-43d2-b9d9-f37bbcb1ce85" />

Submitted to the Firefox Add-ons store! Hopefully, approval happens quickly!

**Comment by Majoramari** [comment]
Glad to see that!

**Comment by richardadonnell** [comment]
Approved!

See link to Firefox extension here: https://addons.mozilla.org/en-US/firefox/addon/upwork-job-scraper-webhook/

Please let me know if you have any issues!

**Comment by Majoramari** [comment]
> Approved!
> 
> See link to Firefox extension here: https://addons.mozilla.org/en-US/firefox/addon/upwork-job-scraper-webhook/
> 
> Please let me know if you have any issues!

Thank you for your work! its working fine for me, I'll try the webhook as soon as I can

**Comment by Majoramari** [comment]
> Thank you for your work! its working fine for me, I'll try the webhook as soon as I can

It's working, fine until now, I'm on Linux, using Zen-browser

<img width="684" height="338" alt="Image" src="https://github.com/user-attachments/assets/4ad788ea-b9de-41fb-b5fc-5babe8682193" />

**Comment by richardadonnell** [comment]
> > Thank you for your work! its working fine for me, I'll try the webhook as soon as I can
> 
> It's working, fine until now, I'm on Linux, using Zen-browser
> 
> <img alt="Image" width="684" height="338" src="https://private-user-images.githubusercontent.com/33487552/566643391-4ad788ea-b9de-41fb-b5fc-5babe8682193.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzM5NzMyODgsIm5iZiI6MTc3Mzk3Mjk4OCwicGF0aCI6Ii8zMzQ4NzU1Mi81NjY2NDMzOTEtNGFkNzg4ZWEtYjlkZS00MWZiLWI1ZmMtNWJhYmU4NjgyMTkzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAzMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMzIwVDAyMTYyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZlYTgyYjAxZjkwOTAwNmI4YTdkNTVmZGZmMmU4YzhmNTU3MTQ5ZDVmYjBmNDYwMTkwNWY0ZDkwMGU0NzBiYjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.eiu_IJqoBlBqKJec2F5Wfpd-VAwHmtaiWDw5YpnHc3c">

That looks like the "test" webhook JSON payload - are you saying you are having issues sending real scraped job data to your webhook URL?

Please let me know a bit more info, and I'll see if I can help resolve it!

**Comment by Majoramari** [comment]
> That looks like the "test" webhook JSON payload - are you saying you are having issues sending real scraped job data to your webhook URL?
> 
> Please let me know a bit more info, and I'll see if I can help resolve it!

The previous one was a test, this one is production

<img width="587" height="485" alt="Image" src="https://github.com/user-attachments/assets/ce7f5ccd-83a4-4955-9e0a-5fc7abd12f2e" />

So no I don't see anything unexpected it's working as epected

**Comment by richardadonnell** [comment]
Awesome man! Glad to hear that! 😎

Cool seeing the notifications formatted on Discord too - nice work! 🚀

Please let me know if you run into any issues, or have any requests - and hopefully this helps land some jobs!! 💵

**Comment by Majoramari** [comment]
> Awesome man! Glad to hear that! 😎
> 
> Cool seeing the notifications formatted on Discord too - nice work! 🚀
> 
> Please let me know if you run into any issues, or have any requests - and hopefully this helps land some jobs!! 💵

Thanks for your hard work. The only issue I anticipate is the Cloudflare captcha.

**Comment by richardadonnell** [comment]
Yeah, unfortunately I can't address the Cloudflare CAPTCHA in this extension. There are some workarounds, but that is outside of the scope of this extension. 😇

I'll go ahead and close this issue now - but please feel free to reach out at any time if you need anything or request any features, etc. 

Happy Hunting! 💸
